x-pack/plugins/monitoring/server/lib/elasticsearch/nodes/get_nodes/get_paginated_nodes.ts

```diff
@@ -61,9 +63,10 @@ export async function getPaginatedNodes(
   const nodes: Node[] = await getNodeIds(req, { clusterUuid }, size);
 
   // Add `isOnline` and shards from the cluster state and shard stats
-  const clusterState = clusterStats?.cluster_state ?? { nodes: {} };
+  const clusterState = clusterStats?.cluster_state ??
+    clusterStats?.elasticsearch?.cluster?.stats?.state ?? { nodes: {} };
   for (const node of nodes) {
-    node.isOnline = !isUndefined(clusterState?.nodes[node.uuid]);
+    node.isOnline = !isUndefined(clusterState?.nodes && clusterState?.nodes[node.uuid]);
```

Comment:
Could we push the grabbing of `.nodes` into the `clusterState` variable creation so that all the unpacking is done in one place?